src/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems

```diff
@@ -1051,6 +1055,10 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\Semaphore.Windows.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\EventWaitHandle.Windows.cs" />
   </ItemGroup>
+  <ItemGroup Condition="$(TargetsWindows) and ('$(FeatureAppX)' == 'true' or '$(EnableWinRT)' == 'true')">
+    <Compile Include="$(MSBuildThisFileDirectory)Internal\Resources\PRIExceptionInfo.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Internal\Resources\WindowsRuntimeResourceManagerBase.cs" />
+  </ItemGroup>
```

Comment:
Moving these two files may or may not be ok. The CoreRT versions are in different namespace and the `Initialize` method has one less argument. Would it be possible to move it and share with ProjectN?

Reply:
Funny story, there's this comment in CoreRT:

```
    // It is important to have WindowsRuntimeResourceManagerBase as regular class with virtual methods and default implementations. 
    // Defining WindowsRuntimeResourceManagerBase as abstract class or interface will cause issues when adding more methods to it 
    // because it'll create dependency between mscorlib and System.Runtime.WindowsRuntime which will require always shipping both DLLs together. 
    // Also using interface or abstract class will not play nice with FriendAccessAllowed.
```

Ironically it seems that these default implementations avoid an error when linking to CoreFX's current System.Runtime.WindowsRuntime. The `Initialize` implementation with 2 arguments has default implementation in CoreRT that does nothing and it is never overridden.

Am I missing something?

Reply:
>  which will require always shipping both DLLs together

This part of the comment was added in an era where we believe that we can ship the framework as 100+ little individually versioned packages. It turned out to be not workable. There .dlls always ship together. You can delete this comment.

> Initialize implementation with 2 arguments has default implementation in CoreRT that does nothing and it is never overridden.

It is overriden in assemblies specific to the .NET Native for UWP that have not been open sourced.

Reply:
> It is overriden in assemblies specific to the .NET Native for UWP that have not been open sourced.

Let me rephrase that questions then.

Is it possible to update the non-open source version to have the same signatures as the version in CoreFX and move them to same namespace? (ie. adding one parameter to `Initialize` method and change namespace from `System.Resources` to `Internal.Resources`)

If yes, is it desirable to share the "interface" files?

If the answer to either of those two questions is "no" then I will just drop this part and move the original definitions in CoreRT to a separate non-shared file and add `Initialize` method there with forward to the 2-argument version.